kolibri/plugins/coach/assets/src/modules/examCreation/handlers.js

```diff
@@ -118,6 +124,56 @@ export function showExamCreationTopicPage(store, params) {
     });
   });
 }
+export function showExamCreationBookamrksPage(store, params) {
+  return store.dispatch('loading').then(() => {
+    const { topicId } = params;
+    const topicNodePromise = ContentNodeResource.fetchModel({ id: topicId });
+    const childNodesPromise = ContentNodeResource.fetchCollection({
+      getParams: {
+        parent: topicId,
+        kind_in: [ContentNodeKinds.TOPIC, ContentNodeKinds.VIDEO, ContentNodeKinds.EXERCISE],
+      },
+    });
+    const loadRequirements = [topicNodePromise, childNodesPromise];
+
+    return Promise.all(loadRequirements).then(([topicNode, childNodes]) => {
+      return filterAndAnnotateContentList(childNodes).then(() => {
+        store.commit('SET_TOOLBAR_ROUTE', {
+          name: PageNames.EXAMS,
+        });
+        return showExamCreationPage(store, {
+          classId: params.classId,
+          bookmarksList: childNodes,
+          pageName: PageNames.EXAM_CREATION_BOOKMARKS,
+          ancestors: [...topicNode.ancestors, topicNode],
+        });
+      });
+    });
+  });
+}
+export function showExamCreationAllBookmarks(store) {
+  return store.dispatch('loading').then(() => {
+    let dataArr = [];
```

Comment:
You don't have to create a variable here outside of the scope of the Promise chain below, but rather can return from one `then` into the next.

```
Promise.resolve(1)
  .then(one => one + 1)
  .then(two => two + 1)  // The value of this function's `two` param is what the function above returned
  .then(three => {
  	return three + 1
   })
  .then(four => console.log(four)); // Outputs 4
```

So in the first `then(/* ... */)` you can gather and `return` the collection of bookmarks, which will be the parameter given to the next `then` clause.